src/lib.rs

```diff
@@ -499,153 +485,217 @@ where
         })
     }
 
-    /// Creates a new message with given data type, schema version and data object.
-    ///
-    /// # Arguments
-    ///
-    /// * data_type -  An Enum instance with static str representation
-    /// * data_schema_version - Version of the data object
-    /// * data - The message data - must be serializable
-    pub fn message<D>(
-        &self,
-        data_type: T,
-        data_schema_version: Version,
-        data: D,
-    ) -> Result<Message<D, T>, MessageError>
+    /// Create a builder for a batch publish construction.
+    pub fn start_publish(&self) -> HedwigPublishBuilder<T, P> {
```

Comment:
Should also mirror `HedwigPublisherBuilder::publish` method here so it doesn't force batch publish API even when publishing a single message.